frappe/erpnext
# feat: Let serialised Items be consumed during Asset Repairs

## Problem

While repairing Assets, users might need to consume Stock Items already in their possession(eg: nuts and bolts). The [Stock Consumption Details](https://docs.erpnext.com/docs/v13/user/manual/en/asset/asset-repair#21-additional-options-when-creating-an-asset-repair) section exists to record such information and decrease stock quantity for the consumed Items. However, this feature can currently only be used for non-serialised Items.

<details>
<summary>Stock Consumption Details section</summary>

![Screenshot 2021-11-25 at 7 25 59 PM](https://user-images.githubusercontent.com/25903035/143454179-93816b7b-2e75-4a04-a015-ae5bcef5c859.png)

</details>

<details>
<summary>Error Message</summary>

The following error is raised when Users attempt to submit Asset Repairs which have consumed Serialized Items:

![Screenshot 2021-11-11 at 6 55 32 PM](https://user-images.githubusercontent.com/25903035/141305891-be4185dd-351a-4ef4-a273-ab6bdb6dc2c0.png)

</details>

## Proposed Change

Allow users to consume Serialized Items during Asset Repairs by:
- Adding a Serial No field
- Adding an _Add Serial No_ button, which prompts the user to enter the Serial Now of their choice

<details>
<summary>Images</summary>

![Screenshot 2021-11-11 at 7 00 05 PM](https://user-images.githubusercontent.com/25903035/141306525-a06fd513-701c-470b-8f99-be9eb6752843.png)

![Screenshot 2021-11-11 at 7 00 57 PM](https://user-images.githubusercontent.com/25903035/141306611-e8821f04-11a6-4e4f-a2de-9ac44e78384f.png)
</details>

<details>
<summary>ToDo</summary>

- [x] Edit [Asset Repair docs](https://docs.erpnext.com/docs/v13/user/manual/en/asset/asset-repair) after this gets merged.

</details>

<details>
<summary>Testing Info</summary>

1. Create a Serialized Item
2. Submit Stock Entry of type Material Receipt for it(or just create Serial No docs)
3. Create an Asset
4. Create an Asset Repair for it
5. Check the _Stock Consumed During Repair_ box
6. Enter the Serialized Item in the Stock Items table

On trying to submit it at this stage, the following error will be raised:
![Screenshot 2021-11-11 at 6 55 32 PM](https://user-images.githubusercontent.com/25903035/141305891-be4185dd-351a-4ef4-a273-ab6bdb6dc2c0.png)

7. Expand the row for the Serialized Item
8. Click on the _Add Serial No_ button and select the Serial No of your choice
9. Save and Submit

This time, you should be able to submit the Asset Repair.

</details>

`no-docs`
